lib/TPP/Bufferize.cpp

```diff
@@ -99,13 +102,21 @@ void Bufferize::runOnOperation() {
     // memrefs are unified in CSE pass, so we can truly remove redundant memcpy.
     passManager.addNestedPass<func::FuncOp>(createCanonicalizerPass());
   }
+  passManager.addPass(bufferization::createDropEquivalentBufferResultsPass());
+
+  if (dealloc) {
+    bufferization::BufferDeallocationPipelineOptions options;
+    bufferization::buildBufferDeallocationPipeline(passManager, options);
+  }
 
+  passManager.addPass(createBufferizationToMemRefPass());
```

Comment:
Is this needed together with oneshot?
If yes, then shouldn't it be placed before the deallocation pass?

Reply:
is needed to convert bufferization operations to memref (i.e., bufferization.dealloc or bufferization.clone).